web/modules/custom/dpl_library_agency/src/Form/GeneralSettingsForm.php

```diff
@@ -148,6 +148,19 @@ public function buildForm(array $form, FormStateInterface $form_state): array {
       '#description' => $this->t('If checked, SMS notifications for patrons will be disabled.'),
     ];
 
+    $form['thresholds'] = [
+      '#type' => 'fieldset',
+      '#title' => $this->t('Thresholds'),
+      '#collapsible' => FALSE,
+      '#collapsed' => FALSE,
+    ];
+
+    $form['thresholds']['threshold_config'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Set thresholds'),
+      '#default_value' => $config->get('threshold_config') ?? '{ "colorThresholds": { "danger": "0", "warning": "6" } }',
+    ];
```

Comment:
The thresholds has probably been a request from the frontenders in your team :)
Which is fine. But I think it is lacking to describe which unit types we are talking about.
So maybe the data structure could look like:
```json
{
  "colorThresholds": {
      "overDueLoans": {
         "danger": "0",
         "warning": "6"
      }  
   }
}
```
or:
```json
{
   "colorThresholds": {
      "danger": {
         "overDueLoans": 0
      },
      "warning": {
         "overDueLoans": 6
      }
   }
}
```
And I think it would be nice with a description on the form field as well.

Reply:
Resolved in Zulip chat.